local-volume/helm/provisioner/templates/provisioner.yaml

```diff
@@ -0,0 +1,76 @@
+{{- $configMapName := .Values.configmap.configMapName }}
+{{- $provisionerNamespace := .Values.common.namespace }}
+{{- $provisionerName := .Values.daemonset.name }}
+{{- $imageFull := .Values.daemonset.image }}
+{{- $imagePullPolicy := .Values.daemonset.imagePullPolicy }}
+{{- $defaultMountDir := .Values.common.defaultMountDir }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ $configMapName }} 
+  namespace: {{ $provisionerNamespace }} 
+data:
+  storageClassMap: |
+#
+# Populating entries with user provided disk discovery directories 
+# and mount directories per storage class.
+#
+{{- range $tmp, $val := .Values.configmap.storageClass }}
+   {{- range $key, $class := $val }}
+      {{ $key }}:     
+         {{- range $val1, $val2 := $class }}
+           {{ $val1 }}: {{ if $val2 }} {{ $val2 }} {{ else }} {{ $defaultMountDir }} {{ end }}
+      {{- end}}
+   {{- end}}
+{{- end}}
+
+---
+
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: {{ $provisionerName }}
+  namespace: {{ $provisionerNamespace }}
+  labels:
+    app: local-volume-provisioner
+spec:
+  selector:
+    matchLabels:
+      app: local-volume-provisioner 
+  template:
+    metadata:
+      labels:
+        app: local-volume-provisioner
+    spec:
+      nodeSelector:
+      containers:
+        - image: "{{ $imageFull }}"
+          imagePullPolicy: {{ $imagePullPolicy | quote }}
+          name: provisioner 
+          securityContext:
+            privileged: true
+          env:
+          volumeMounts:
+            - mountPath: /etc/provisioner/config 
+              name: provisioner-config
+              readOnly: true
+#
+# Populating entries with user provided disk discovery directories 
+#
+{{- range $tmp, $val := .Values.configmap.storageClass }}
+   {{- range $key, $class := $val }}
+            - mountPath: {{ if $class.mountDir }} {{ $class.mountDir }} {{ else }} {{ $defaultMountDir }} {{ end }}
```

Comment:
Using defaultMountDir doesn't work here.  The mount dir has to be unique for each storageclass config.  So you need to generate a mountdir with some either some random prefix, or have it incorporate the hostDir somehow.

Reply:
random prefix will not work as it has to configmap and daemonset are generated separately and this "random" prefix must be the same between them. Please check the alternative solution.